pkg/migration/controllers/migration.go

```diff
@@ -1564,6 +1575,82 @@ func (m *MigrationController) getParsedLabels(
 	return m.getParsedMap(labels, clusterPair)
 }
 
+// updateStorageClassOnPV updates StorageClass on PV object
+// StorageClass is created on destination if it doesn't exist
+func (m *MigrationController) updateStorageClassOnPV(
+	migration *stork_api.Migration,
+	objects []runtime.Unstructured,
+	clusterPair *stork_api.ClusterPair,
+) error {
+	remoteClient, err := m.getRemoteClient(migration)
+	if err != nil {
+		return err
+	}
+	// Get a list of StorageClasses from destination
+	destStorageClasses, err := remoteClient.adminClient.StorageV1().StorageClasses().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	destScExists := make(map[string]bool)
+	for _, sc := range destStorageClasses.Items {
+		destScExists[sc.Name] = true
+	}
+
+	for _, obj := range objects {
+		metadata, err := meta.Accessor(obj)
+		if err != nil {
+			return err
+		}
+		if obj.GetObjectKind().GroupVersionKind().Kind != "PersistentVolume" {
+			continue
+		}
+		destPV, err := remoteClient.adminClient.CoreV1().PersistentVolumes().Get(context.TODO(), metadata.GetName(), metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		if scName, ok := destPV.Annotations[resourcecollector.CurrentStorageClassName]; ok {
+			// Create StorageClass on destination if it doesn't exist
+			if _, ok := destScExists[scName]; !ok {
+				// Get StorageClass from source
+				sc, err := storkcache.Instance().GetStorageClass(scName)
+				if err != nil {
+					return err
+				}
+				sc.UID = ""
+				sc.ResourceVersion = ""
+				log.MigrationLog(migration).Infof("Applying %v %v", sc.Kind, sc.Name)
+				for retries := 0; retries < maxApplyRetries; retries++ {
+					_, err = remoteClient.adminClient.StorageV1().StorageClasses().Create(context.TODO(), sc, metav1.CreateOptions{})
+					if err == nil {
```

Comment:
Can we add err == nil || errors.IsAlreadyExists(err) ?

Reply:
Also what will happen if there exists another storageclass with same name but different parameters or even provisioners? this case is very unlikely though.

Reply:
1. We are only creating the SC if it doens't exist. So I think we will never get "IsAlreadyExists". However, it still is logical to handle this condition. Updated as you mentioned.
2. We don't want to migrate SC if it already exists on destination. So if Sc exists with different params, we will not migrate the SC. We would still update the SC name to the PV.